lib/default.ml

```diff
@@ -38,27 +44,73 @@ module Str = struct
       Raise.Default.errorf
         ~loc:ptype_loc
         "can't derive default for an abstract type without a manifest"
-    | Some typ -> expr_from_core_type ~loc typ
+    | Some typ -> expr_from_core_type_exn ~loc typ
 
   let field_binding ~loc {pld_name; pld_type; _} =
+    let open Util.Result_ in
     let lident = {txt = Lident pld_name.txt; loc} in
-    let expr = expr_from_core_type ~loc pld_type in
+    expr_from_core_type ~loc pld_type >|= fun expr ->
     (lident, expr)
 
   let value_expr_from_labels ~loc labels =
+    let open Util.Result_ in
     let field_bindings = List.map (field_binding ~loc) labels in
+    Util.List_.all_ok field_bindings >|= fun field_bindings ->
     Ast_builder.Default.pexp_record ~loc field_bindings None
 
+  let value_expr_from_labels_exn ~loc labels =
+    Loc_err.ok_or_raise @@ value_expr_from_labels ~loc labels
+
+  let value_expr_from_ctr_tuple ~loc types =
+    let open Util.Result_ in
+    let expr_list = List.map (expr_from_core_type ~loc) types in
+    match expr_list with
+    | [] -> Ok None
+    | [expr] -> expr >|= fun expr -> Some expr
+    | _ ->
+      Util.List_.all_ok expr_list >|= fun expr_list ->
+      Some (Ast_builder.Default.pexp_tuple ~loc expr_list)
+
+  let value_expr_from_ctor ~loc {pcd_name = {txt = ctr_name; _}; pcd_args; _} =
```

Comment:
What do `ctr` and `ctor` mean (and could they be named a bit more descriptively)?

Reply:
They stand for `constructor`, I'll replace them!